### DISCUSSION TITLE
The impact of too much bindings ?(the impact of too much background tasks)

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3325] [Q&A]
When using a large number of bindings, how much impact will it have on the program's performance? Will the resources consumed by background refreshes increase as the number of bindings grows?

**Comment by python-and-novella** [comment]
Yes.You could hack source code.
Nicegui uses async to refresh data binding,everytime you fresh page after binding a lot of values,app will wait more time ( depend on the number of values binded)

**Reply by python-and-novella** [reply]
When binding costs too long time.It will appear.
![image](https://github.com/zauberzeug/nicegui/assets/16256910/2ff6368e-4b8e-4c47-a43c-5e3ed613b800)

**Reply by v479038280** [reply]
If ex4nicegui is a better solution

**Reply by python-and-novella** [reply]
Have you tested yet?

**Reply by v479038280** [reply]
I think ex4nicegui don't have impact on the performance , according to it's documents.I don't know how to test the performance. But but binding is more convenient for use 。Sorry for the late reply, my English is not very good, and I'm a bit slow with wording

**Comment by falkoschindler** [comment]
This comment might shed some light on this subject: https://github.com/zauberzeug/nicegui/discussions/1975#discussioncomment-7514899.
I just created PR #3333 to add it to the documentation.
